### PR TITLE
Move gltf to a module for future proposals.

### PR DESCRIPTION
**Pull request by fire**
Support for proposal https://github.com/godotengine/godot-proposals/issues/885

The goal of this pr is to try to make my work in progress https://github.com/godot-extended-libraries/gltf mergable. Also, this is how the new FBX importer works, so having the gltf2 importer be a module makes sense. The gltf2 doesn't have any internal dependencies so it can be moved out.

Thoughts on moving the collada and the escn importers out too?

### CODE REVIEW

File: modules/gltf/config.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def can_build(env, platform):
+    return True
```
Comment (akien-mga) [inline modules/gltf/config.py:2]: It shouldn't build if `tools=no`.

File: modules/gltf/SCsub [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python
+
+Import("env")
+Import("env_modules")
+
+env_gltf = env_modules.Clone()
+env_gltf.Prepend(CPPPATH=["."])
```
Comment (akien-mga) [inline modules/gltf/SCsub:7]: Why is that needed?

File: modules/gltf/register_types.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "register_types.h"
+
+#ifndef _3D_DISABLED
+#ifdef TOOLS_ENABLED
```
Comment (akien-mga) [inline modules/gltf/register_types.cpp:34]: Those should enclose the whole file.

**Comment by akien-mga** [comment]
I'm not sure I see much point in doing this. But let's see what @reduz thinks about it.

File: modules/gltf/register_types.cpp (hunk 3 above)
Comment (Xrayez) [reply modules/gltf/register_types.cpp:34]: Or just compile only when required via `SCSub` itself if `not env["disable_3d"]` and `env["tools"]`, could improve readability.

Comment (fire) [reply modules/gltf/register_types.cpp:34]: My goal is to make gltf2 work at runtime, but that's not this pr.

**Comment by RevoluPowered** [comment]
I can make a PR and move collada to a module too if you want @akien-mga we could do them all?

**Comment by akien-mga** [comment]
You should see with @reduz what he thinks of the proposal first.

**Comment by fire** [comment]
Since there has been no movement, I'll work on implementing the whole proposal https://github.com/godotengine/godot/pull/34193. I was hoping this would be an easier merge.
